lib/utils.py

```diff
@@ -389,10 +389,50 @@ def ping_host(ip):
     else:
         return False
 
-
 def is_file_executable(file):
     output = subprocess.check_output(['file', file])
     if ("Installer" in output) or ("executable" in output):
         return True
     else:
         return False
+
+def add_domain_blacklist(domain):
+    if domain_blacklist_check_if_exists(domain):
+        fout = open('/etc/dnsmasq.block', 'a')
+        fout.write('\n' + '127.0.0.1' + '\t' + domain)
+        fout.close()
+        log.debug('FG-INFO: Domain ' + domain + ' added to user blacklist')
+    else:
+        log.debug('FG-INFO: Something went wrong when adding Domain ' + domain + ' to user blacklist')
+    utils.restart_dnsmasq()
+
+def domain_blacklist_check_if_exists(domain):
```

Comment:
This function returns False if the domain it's present in the blacklist when it should return True. Consider this alternative instead:
```
def domain_blacklist_check_if_exists(domain):
    f = open("/etc/dnsmasq.block", "r")
    d = f.readlines()
    f.close()
    for line in d:
        if domain in line:
            return True
    return False
```